sycl/doc/extensions/ExtendedAtomics/SYCL_INTEL_extended_atomics.asciidoc

```diff
@@ -113,389 +115,404 @@ In SYCL 1.2.1, support for floating-point types is limited to the +load+, +store
 
-  /* Available only when: T != float */
-  T fetch_min(T operand, memory_order memoryOrder =
-    memory_order::relaxed);
+The effects of a call to +atomic_fence+ depend on the value of the +order+ parameter:
 
-  /* Available only when: T != float */
-  T fetch_max(T operand, memory_order memoryOrder =
-    memory_order::relaxed);
-};
-} // namespace sycl
-} // namespace cl
-----
+- `relaxed`: No effect
+- `acquire`: Acquire fence
+- `release`: Release fence
+- `acq_rel`: Both an acquire fence and a release fence
+- `seq_cst`: A sequentially consistent acquire and release fence
 
-==== To:
+==== Sample  Header
 
 [source,c++]
 ----
 namespace cl {
 namespace sycl {
-enum class memory_order : int {
-  relaxed
-};
 namespace intel {
-template <typename T, access::address_space addressSpace =
-  access::address_space::global_space>
-class atomic {
+enum class memory_order : /* unspecified */ {
+  relaxed, acquire, release, acq_rel, seq_cst
+};
+inline constexpr memory_order memory_order_relaxed = memory_order::relaxed;
+inline constexpr memory_order memory_order_acquire = memory_order::acquire;
+inline constexpr memory_order memory_order_release = memory_order::release;
+inline constexpr memory_order memory_order_acq_rel = memory_order::acq_rel;
+inline constexpr memory_order memory_order_seq_cst = memory_order::seq_cst;
+
+// Exposition only
+template <memory_order ReadModifyWriteOrder>
+struct memory_order_traits;
+
+template <>
+struct memory_order_traits<memory_order::relaxed> {
+  static constexpr memory_order read_order = memory_order::relaxed;
+  static constexpr memory_order write_order = memory_order::relaxed;
+};
+
+template <>
+struct memory_order_traits<memory_order::acq_rel> {
+  static constexpr memory_order read_order = memory_order::acquire;
+  static constexpr memory_order write_order = memory_order::release;
+};
+
+template <>
+struct memory_order_traits<memory_order::seq_cst> {
+  static constexpr memory_order read_order = memory_order::seq_cst;
+  static constexpr memory_order write_order = memory_order::seq_cst;
+};
+
+template <typename T, memory_order DefaultOrder, access::address_space Space>
+class atomic_ref {
  public:
 
-  atomic(multi_ptr<T, addressSpace> ptr);
-  atomic(const atomic&);
-  atomic& operator=(const atomic&) = delete;
+  using value_type = T;
+  static constexpr size_t required_alignment = /* implementation-defined */;
+  static constexpr bool is_always_lock_free = /* implementation-defined */;
+  static constexpr memory_order default_read_order = memory_order_traits<DefaultOrder>::read_order;
+  static constexpr memory_order default_write_order = memory_order_traits<DefaultOrder>::write_order;
+  static constexpr memory_order default_read_modify_write_order = DefaultOrder;
+
+  bool is_lock_free() const noexcept;
 
-  bool is_lock_free() const;
+  explicit atomic_ref(T&);
```

Comment:
The specification for standard `atomic_ref` says that it can't throw anything, but doesn't have `noexcept` here (see http://eel.is/c++draft/atomics.ref.generic).  I don't understand this, but I think we want to be aligned with the C++ definition as much as possible.

Reply:
Thanks for explanation, I don't have any further objections here